# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/agnostic/bytestreams/H265.h
# Commit: 4989219e22b6
# Full Hash: 4989219e22b63966518e1b7fa107b530d6e7982f
# Author: alwu <alwu@mozilla.com>
# Date: 2023-10-11 09:22:03
# Regressor Bug: 1852333
# File Overlap Count: 4
# Description:
#   Bug 1852333 - part2 : detect config change for HEVC stream. r=media-playback-reviewers,padenot
#   
#   This patch detects inband change for the HEVC playback so that we can
#   recreate decoder if the new config is different from the previous one.
#   
# ==============================================================================

diff -r 81fca2c97c24 -r 4989219e22b6 dom/media/platforms/agnostic/bytestreams/H265.h
--- a/dom/media/platforms/agnostic/bytestreams/H265.h	Tue Oct 10 17:20:27 2023 +0000
+++ b/dom/media/platforms/agnostic/bytestreams/H265.h	Tue Oct 10 17:20:27 2023 +0000
@@ -36,6 +36,7 @@
 class H265NALU final {
  public:
   H265NALU(const uint8_t* aData, uint32_t aByteSize);
+  H265NALU() = default;
 
   // Table 7-1
   enum NAL_TYPES {
@@ -183,6 +184,9 @@
 struct H265SPS final {
   H265SPS() = default;
 
+  bool operator==(const H265SPS& aOther) const;
+  bool operator!=(const H265SPS& aOther) const;
+
   // Syntax elements.
   uint8_t sps_video_parameter_set_id = {};
   uint8_t sps_max_sub_layers_minus1 = {};
@@ -255,6 +259,7 @@
       const mozilla::MediaByteBuffer* aExtraData);
 
   uint8_t NALUSize() const { return lengthSizeMinusOne + 1; }
+  uint32_t NumSPS() const;
   bool HasSPS() const;
 
   uint8_t configurationVersion;
@@ -292,6 +297,14 @@
   static Result<H265SPS, nsresult> DecodeSPSFromSPSNALU(
       const H265NALU& aSPSNALU);
 
+  // Extract SPS and PPS NALs from aSample by looking into each NALs.
+  static already_AddRefed<mozilla::MediaByteBuffer> ExtractHVCCExtraData(
+      const mozilla::MediaRawData* aSample);
+
+  // Return true if both extradata are equal.
+  static bool CompareExtraData(const mozilla::MediaByteBuffer* aExtraData1,
+                               const mozilla::MediaByteBuffer* aExtraData2);
+
  private:
   // Return RAW BYTE SEQUENCE PAYLOAD (rbsp) from NAL content.
   static already_AddRefed<mozilla::MediaByteBuffer> DecodeNALUnit(